packages/melody-hooks/src/component.js

```diff
@@ -90,11 +91,11 @@ Object.assign(Component.prototype, {
      * @param {*} props new properties
      */
     apply(props) {
-        if (shallowEquals(props, this.props)) {
-            return;
```

Comment:
I removed the early return from `apply` and just set the `isPropsDirty` property. This way we always call `this.enqueuComponent`. If props haven't changed but we have a pending state change, the state change will also be proceeded and we don't need to revisit the component.